platforms/jvm/scala/src/main/java/org/gradle/api/internal/tasks/scala/DaemonScalaCompiler.java

```diff
@@ -83,8 +83,8 @@ protected DaemonForkOptions toDaemonForkOptions(T spec) {
         MinimalScalaCompileOptions compileOptions = spec.getScalaCompileOptions();
         MinimalScalaCompilerDaemonForkOptions forkOptions = compileOptions.getForkOptions();
         JavaForkOptions javaForkOptions = new MinimalCompilerDaemonForkOptionsConverter(forkOptionsFactory).transform(mergeForkOptions(javaOptions, forkOptions));
-        javaForkOptions.setWorkingDir(daemonWorkingDir);
-        javaForkOptions.setExecutable(spec.getJavaExecutable().getAbsolutePath());
+        javaForkOptions.getWorkingDir().set(daemonWorkingDir);
+        javaForkOptions.getExecutable().set(spec.getJavaExecutable().getAbsolutePath());
```

Comment:
🤔 Keeping `executable` a `String` instead of making it a `RegularFile` feels just... wrong. I know we said we are not fixing further problems outside of making eager properties lazy, but it still feels like we should use the opportunity here. I'm conflicted...